### PR TITLE
Feat: Budget Resource (for project)

### LINKED ISSUE #703: Feat: Budget Resource (for project)
We created an option to define an **budget for a project** (optionally to be defined later for org/env as well) 
image

Resource name: `env_project_budget`

Fields:
```
  export interface Budget {
    id?: string;
    projectId: string;
    amount: number;
    timeframe: 'WEEKLY' | 'MONTHLY' | 'QUARTERLY' | 'YEARLY';
    thresholds: number[];
  }
```

It has the following API:

1. [DELETE](https://docs.env0.com/reference/cost-remove-project-budget)
2. [GET](https://docs.env0.com/reference/cost-get-project-budget)
3. [PUT](https://docs.env0.com/reference/cost-upsert-project-budget) (upsert)

### PR DESCRIPTION
**Pull request by TomerHeber**
### Issue & Steps to Reproduce / Feature Request
resolves #703 

### Solution

1. Added the new resource schema.
2. Added acceptance tests.
3. Added integration tests.
4. Added an example.
5. Added support for slice of ints.

### CODE REVIEW

File: env0/test_helpers.go [hunk 1]
```diff
@@ -43,23 +43,30 @@ func hclConfigCreate(source TFSource, resourceType string, resourceName string,
 	for key, value := range fields {
 		intValue, intOk := value.(int)
 		boolValue, boolOk := value.(bool)
-		arrayValues, arrayOk := value.([]string)
+		arrayStrValues, arrayStrOk := value.([]string)
+		arrayIntValues, arrayIntOk := value.([]int)
```
Comment (TomerHeber) [inline env0/test_helpers.go:47]: added support for array of ints.

File: env0/utils.go [hunk 2]
```diff
@@ -201,6 +201,8 @@ func readResourceDataSliceEx(field reflect.Value, resources []interface{}) error
 		switch elemType.Kind() {
 		case reflect.String:
 			val = reflect.ValueOf(resource.(string))
+		case reflect.Int:
```
Comment (TomerHeber) [inline env0/utils.go:204]: Added support for an array of ints.

File: env0/test_helpers.go [hunk 3]
```diff
@@ -43,23 +43,30 @@ func hclConfigCreate(source TFSource, resourceType string, resourceName string,
...
 			arrayValueString := ""
-			for _, arrayValue := range arrayValues {
+			for _, arrayValue := range arrayStrValues {
 				arrayValueString += "\"" + arrayValue + "\","
 			}
 			arrayValueString = arrayValueString[:len(arrayValueString)-1]
 
 			hclFields += fmt.Sprintf("\n\t%s = [%s]", key, arrayValueString)
-		}
-		if !intOk && !boolOk && !arrayOk {
```
Comment (TomerHeber) [inline env0/test_helpers.go:62]: not needed... instead I use "if else" (before it was just "if" with no else).

File: env0/resource_project_budget_test.go [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+					}),
+					ExpectError: regexp.MustCompile("must be one of: WEEKLY, MONTHLY, QUARTERLY, YEARLY"),
+				},
+			},
+		}
+
+		runUnitTest(t, testCase, func(mock *client.MockApiClientInterface) {})
+	})
+
+	t.Run("Detect drift", func(t *testing.T) {
```
Comment (yarivg) [inline env0/resource_project_budget_test.go:106]: how does it detect a drift? I didn't understand this test

**Review by yarivg — APPROVED** [review]
LGTM 👍

File: env0/resource_project_budget_test.go (hunk 4 above)
Comment (TomerHeber) [reply env0/resource_project_budget_test.go:106]: `mock.EXPECT().ProjectBudget(projectBudget.ProjectId).Times(1).Return(nil, &client.NotFoundError{}),`

It returns a not found error.

**Pull request merged**